### PR TITLE
Enable RES on New ModMail pages

### PR DESCRIPTION
**Pull request by bradgrafelman**
- Handle New ModMail as a new AppType ("nmm") and add new PageType's
- Always assume www.reddit.com as host when fetching announcements
- Pre-allocate a new NightMode body class name ("res-nmm-nightmode")

Ref #4068 and #4069

Tested in browser: Chrome v67

### CODE REVIEW

File: lib/utils/location.js [hunk 1]
```diff
@@ -51,12 +53,14 @@ export const execRegexes: { [string]: (path: string) => ?string[] } = {
 	},
 };
 
-export type PageType = 'wiki' | 'search' | 'profile' | 'profile2x' | 'profileCommentsPage' | 'stylesheet' | 'modqueue' | 'subredditAbout' | 'comments' | 'commentsLinklist' | 'liveThread' | 'inbox' | 'submit' | 'account' | 'prefs' | 'linklist';
-export type AppType = 'r2' | 'd2x';
+export type PageType = 'wiki' | 'search' | 'profile' | 'profile2x' | 'profileCommentsPage' | 'stylesheet' | 'modqueue' | 'subredditAbout' | 'comments' | 'commentsLinklist' | 'liveThread' | 'inbox' | 'submit' | 'account' | 'prefs' | 'linklist' | 'newModMail' | 'composeNewModMail';
+export type AppType = 'r2' | 'd2x' | 'nmm';
```
Comment (andytuba) [inline lib/utils/location.js]: How about `mm2x`?  Internally at Reddit, "new modmail / modmail beta" is part of the "2x" family, i.e. Node-served React webapps.

File: lib/utils/location.js [hunk 2]
```diff
@@ -74,6 +78,9 @@ const appPageTypes: { [AppType]: {| default?: PageType, pageTypes: PageType[] |}
 	d2x: {
 		pageTypes: ['profile2x', 'profileCommentsPage'],
 	},
+	nmm: {
+		pageTypes: ['newModMail', 'composeNewModMail'],
```
Comment (andytuba) [inline lib/utils/location.js]: per https://github.com/honestbleeps/Reddit-Enhancement-Suite/pull/4790/files#r194815907

How about `modmail2x` and `modmail2xCompose`?

File: lib/utils/location.js [hunk 3]
```diff
@@ -39,6 +39,8 @@ export const regexes: { [string]: RegExp } = {
 	domain:           /^\/domain\/([\w\.]+)(?:\/|$)/i,
 	composeMessage:   /^\/(?:r\/([\w\.\+]+)\/)?message\/compose(?:\/|$)/i,
 	liveThread:       /^\/live\/(?!create(?:\/|$))([a-z0-9]+)(?:\/|$)/i,
+	newModMail:       /^\/mail\/(?:all|new|inprogress|archived|highlighted|mod|notifications)(?:\/([0-9a-z]+))?\/?$/i,
```
Comment (andytuba) [inline lib/utils/location.js]: I haven't yet explicitly double-checked this but it looks pretty comprehensive and safe.

**Comment by bradgrafelman** [comment]
The style suggestions are very much appreciated (I'm severely lacking in terminology/naming schemes in RES or Reddit). I'm working on implementing them now.

File: lib/utils/location.js [hunk 4]
```diff
@@ -74,6 +78,9 @@ const appPageTypes: { [AppType]: {| default?: PageType, pageTypes: PageType[] |}
 	d2x: {
 		pageTypes: ['profile2x', 'profileCommentsPage'],
 	},
+	mm2x: {
+		pageTypes: ['modmail2x', 'modmail2x'],
```
Comment (mc10) [inline lib/utils/location.js]: I believe the second type here should be `modmail2xCompose`.

File: chrome/manifest.json [hunk 5]
```diff
@@ -31,7 +31,6 @@
 			"https://*.reddit.com/*"
 		],
 		"exclude_matches": [
-			"https://mod.reddit.com/*",
```
Comment (BenMcGarry) [inline chrome/manifest.json]: Can you update this change across all browser manifests?
Edge - https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/edge/manifest.json
Firefox - https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/master/firefox/manifest.json

**Pull request closed without merging**